Notification Service/NotificationService.swift

```diff
@@ -38,19 +40,30 @@ class NotificationService: UNNotificationServiceExtension {
         Task {
             await self.initializeParse()
 
-            guard let chatClient = self.getConfiguredChatClient() else { return }
+            guard let chatClient = await self.getConfiguredChatClient(),
+                  let mutableContent = request.content.mutableCopy() as? UNMutableNotificationContent else {
+
+                      contentHandler(request.content)
+                      return
+                  }
 
-            await self.updateContent(with: request,
-                                     client: chatClient,
-                                     contentHandler: contentHandler)
+
+            await self.initializeNotificationService(with: mutableContent, client: chatClient)
+            await self.updateInterruptionLevel(of: mutableContent)
+            await self.updateBadgeCount(of: mutableContent)
+
+            let finalContent = await self.finalizeContent(mutableContent)
+            contentHandler(finalContent)
```

Comment:
I refactored this class to use be more async friendly. The functions don't call each other or pass around the completion handler. It's always executed in this order.